python/vineyard/data/dataset.py

```diff
@@ -0,0 +1,66 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Alibaba Group Holding Limited.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from vineyard._C import ObjectMeta
+from torch.utils.data import Dataset
+from .utils import from_json, to_json
+
+
+def dataset_builder(client, value, builder):
+    meta = ObjectMeta()
+    meta['typename'] = 'vineyard::DataSet'
+    meta['num'] = to_json(len(value))
+    for i in range(len(value)):
+        data, label = value[i]
+        meta.add_member(f'data_{i}_', builder.run(client, data))
+        meta.add_member(f'label_{i}_', builder.run(client, label))
+    return client.create_metadata(meta)
```

Comment:
How this dataset interactive with things like `DataFrame` or `Tensor`?

Shouldn't we build `DataSet` from `DataFrame` or `Tensor`, rather than create a new class? How could the new datatype interact with other applications bulid upon vineyard, like pandas as well as GAE?

Reply:
The Dataset type is indeed a composition type from dataframe and tensor. But the structure itself deserves a type.

Reply:
The how could map data generated by other engines to `Dataset` and shared with pytorch/tf/mxnet?